src/Microsoft.Diagnostics.DebugServices.Implementation/Runtime.cs

```diff
@@ -96,9 +99,36 @@ public Version RuntimeVersion
             }
         }
 
-        public string GetDacFilePath()
+        public string GetCDacFilePath()
         {
-            _dacFilePath ??= GetLibraryPath(DebugLibraryKind.Dac);
+            if (_cdacFilePath is null)
+            {
+                if (_settingsService.UseContractReader || _settingsService.ForceUseContractReader)
+                {
+                    _cdacFilePath = GetLibraryPath(DebugLibraryKind.CDac);
+                }
+            }
+            return _cdacFilePath;
+        }
+
+        public string GetDacFilePath(out bool verifySignature)
+        {
+            verifySignature = false;
+            if (_settingsService.ForceUseContractReader)
+            {
+                return GetCDacFilePath();
+            }
+            if (_dacFilePath is null)
+            {
+                if (_dacFilePath is null)
```

Comment:
Probably one too many null checks for this?

Reply:
Done